sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegen.scala

```diff
@@ -284,7 +284,14 @@ case class WholeStageCodegen(plan: CodegenSupport, children: Seq[SparkPlan])
         ${ctx.declareAddedFunctions()}
 
         protected void processNext() throws java.io.IOException {
-          ${code.trim}
+          try {
+            ${code.trim}
+          } catch (final Throwable e) {
+            org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(this.getClass());
+            logger.error("The method processNext() is generated for " +
+                         "${plan.origin.callSite.getOrElse("unknown")}");
+            throw e;
+          }
```

Comment:
There are potentially exception-throwing instructions (PEIs) such as `invokevirtual` (e.g. ```inputadapter_row.isNullAt```). In the future, there would be PEIs when we generate code for more complex SQL operations.

For ```Logging```, let me investigate it.